### PR TITLE
schema changes for widget config

### PR DESCRIPTION
**Pull request by SuddenDevelopment**
public/config/new_config.json has an example config in the new format
the readme in the same directory has been largely replaced with the type, links, properties used in this new format

### CODE REVIEW

File: public/config/README_configuration.md [hunk 1]
```diff
@@ -29,81 +29,65 @@ The configuration file has four main sections and then several key-value pairs.
 The definition of the widgets placed on the page. The "widgets" property is
 an array of objects uniquely identified with the properties "type" and "id".
 
-#### widget CSS properties
+widget types:
+        socketValue: displayed float value
```
Comment (billmania) [inline public/config/README_configuration.md:33]: Prefixing every type with "socket" doesn't seem to add any information value. It also implies these types only work with "socket", and no other mechanism. There are other widget types we'll implement, such as "Label", which doesn't have any robot data.

File: public/config/README_configuration.md [hunk 2]
```diff
@@ -29,81 +29,65 @@ The configuration file has four main sections and then several key-value pairs.
...
+        socketButton: emit a value when clicked
+        socketState: display boolean state
```
Comment (billmania) [inline public/config/README_configuration.md:35]: Our user community got used to these widgets being called "Indicator". In addition, I think Indicator may be more intuitive to a high school student.

File: public/config/README_configuration.md [hunk 3]
```diff
@@ -29,81 +29,65 @@ The configuration file has four main sections and then several key-value pairs.
...
-                               the x and y values.
+#### widget properties
 
+[CSS reference](https://www.w3schools.com/cssref/index.php)
+[widget lib](https://jqueryui.com/widget/)
+[position lib](https://jqueryui.com/position/)
+[button lib](https://jqueryui.com/button/)
+[slider lib](https://jqueryui.com/slider/)
+
+the widget properties have the following sections:
```
Comment (billmania) [inline public/config/README_configuration.md:47]: I like this summary.

File: public/config/README_configuration.md [hunk 4]
```diff
@@ -29,81 +29,65 @@ The configuration file has four main sections and then several key-value pairs.
...
+* label: string displayed on the widget header
+
+* position.my: string defining x,y of widget to position with respect to the parent
+* position.at: string defining x,y of parent to position widget
+
+* format.prefix: (value) string prepended to the value before displaying
+* format.suffix: (value) string appended to the value before displaying
+* format.precision: (value) number of decimal places to display for values 
+
+* format.trueText: (state) string displayed when state is true
```
Comment (billmania) [inline public/config/README_configuration.md:64]: I find this addition useful too, to clearly describe for which this attribute is relevant.
For consistency, if it's described as "Slider" in the list above, which she keep the case consistent here.

File: public/config/new_config.json [hunk 5]
```diff
@@ -0,0 +1,114 @@
+{
+    "widgets": [
+        {
+            "type": "socketState",
```
Comment (billmania) [inline public/config/new_config.json:4]: See the comment above about removing "socket".

File: public/config/new_config.json [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+                "min": "0",
+                "max": "180",
+                "step": "1",
+                "default": "90",
+                "orientation": false
+            },
+            "data": {
+                "topic": "servos",
+                "topicDirection": "publish",
+                "topicAttribute": "name",
```
Comment (billmania) [inline public/config/new_config.json:94]: This attribute name is incorrect. We can review how best to handle slider data for the ServoAngles topicType, then I'll make simplifications in the NodeJS code and you can implement the browser code.

File: public/config/README_configuration.md (hunk 2 above)
Comment (SuddenDevelopment) [reply public/config/README_configuration.md:35]: done

File: public/config/README_configuration.md (hunk 1 above)
Comment (SuddenDevelopment) [reply public/config/README_configuration.md:33]: removed prefix

File: public/config/new_config.json (hunk 6 above)
Comment (SuddenDevelopment) [reply public/config/new_config.json]: updated to an array

**Review by billmania — APPROVED** [review]

**Pull request merged**